app/(dashboard)/reports/[id]/report-item.tsx

```diff
@@ -1,57 +1,141 @@
-      <TableCell className="hidden md:table-cell">{item.article}</TableCell>
-      <TableCell className="hidden max-w-64 md:table-cell truncate">
-        {item.serialNo}
-      </TableCell>
-      <TableCell className="hidden md:table-cell">
-        {item.warrantyType.name}
-      </TableCell>
-      <TableCell className="hidden md:table-cell">
-        {item.serviceLevelType.name}
-      </TableCell>
-      <TableCell>
-        {item.dateOut ? item.dateOut.toLocaleDateString('en-US') : ''}
-      </TableCell>
-      <TableCell>{item.comments}</TableCell>
-      <TableCell>
-        <DropdownMenu>
-          <DropdownMenuTrigger asChild>
-            <Button aria-haspopup="true" size="icon" variant="ghost">
-              <MoreHorizontal className="h-4 w-4" />
-              <span className="sr-only">Toggle menu</span>
-            </Button>
-          </DropdownMenuTrigger>
-          <DropdownMenuContent align="end">
-            <DropdownMenuLabel>Actions</DropdownMenuLabel>
-            <DropdownMenuItem>Edit</DropdownMenuItem>
-            <DropdownMenuItem>
-              <form action={() => {}}>
-                <input type="hidden" name="id" value={item.id} />
-                <button type="submit">Delete</button>
-              </form>
-            </DropdownMenuItem>
-          </DropdownMenuContent>
-        </DropdownMenu>
-      </TableCell>
-    </TableRow>
-  );
-};
+'use client';
+
+import { TableRow } from '@/components/ui/table';
+import { ReportItemWithNames } from '@/lib/data/reportItems';
+import ReportItemCell from './report-item-cell';
+import ReportItemEditMenu from './report-item-edit-menu';
+import React from 'react';
+import { deleteReportItem, saveReportItem } from './actions';
+import ReportItemDropdown from './report-item-dropdown';
+import ReportItemDatePicker from './report-item-date-picker';
+
+const ReportItem = ({
+  item,
+  brands,
+  serviceLevelTypes,
+  warrantyTypes
+}: {
+  item: ReportItemWithNames;
+  brands: { id: number; name: string }[];
+  serviceLevelTypes: { id: number; name: string }[];
+  warrantyTypes: { id: number; name: string }[];
+}) => {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [editedItem, setEditedItem] = React.useState(item);
+
+  const handleChange =
+    (field: keyof ReportItemWithNames) =>
+    (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      setEditedItem({ ...editedItem, [field]: e.target.value });
+    };
+  const findBrandByID = (id: number): string => {
+    return brands.find((brand) => brand.id === id)?.name || '';
+  };
+  const findServiceByID = (id: number): string => {
+    return (
+      serviceLevelTypes.find((serviceLevelType) => serviceLevelType.id === id)
+        ?.name || ''
+    );
+  };
+  const findWarrantyByID = (id: number): string => {
+    return (
+      warrantyTypes.find((warrantyType) => warrantyType.id === id)?.name || ''
+    );
+  };

```

Comment:
These are very repetitive. Could they be consolidated? Maybe something like this:

```suggestion
const findNameById = (id: number, items: { id: number; name: string }[]) =>
    items.find((item) => item.id === id)?.name || '';
```

Reply:
Implemented